src/consensus/vote.rs

```diff
@@ -58,6 +58,9 @@ pub(crate) enum Vote {
         message: Box<PlainMessage>,
         proof_chain: SectionProofChain,
     },
+
+    // Voted to concensus whether new node shall be allowed to join
+    JoinFlag(bool),
```

Comment:
I don't think we need to vote for this. Just setting the join flag individually by each node should be sufficient, no?

Reply:
This is to reduce the chance of accumulating mis-flag and causing issue.
The vote is also used to notify a newly promoted elder of what flag value shall be.

Reply:
We should probably discuss this with the sn-node team to understand how exactly will this be used, but my impression is that the elders will track all the adults in the section and when they detect the average storage capacity (or some other resource) is too low, they will flip the flag to true so the section starts accepting new nodes. All the elders should detect this more or less at the same time, so eventually all the elders will flip the flag equally. But even if there is inconsistency, the majority value will decide what happens. So if the majority has it set to `true`, then the section will accept new nodes, otherwise they won't. Seems good enough already. I'd just like to keep things simple if possible.

Reply:
This is correct. The issue in routing is adding a node will mean Infants (joining queue). As infants don't add to sn_node resources then we need to do more here. 
1 suggestion is when this Flag is set we increment Infants age + promote them to Adults? (promote here is probably OK and no need to change the name as Infants will not be held in Elders or Adult containers (we do need separate containers though, but that's more obvious))